code/Data exploration.R

```diff
@@ -78,5 +117,49 @@ data <- data %>%mutate(WEEKDAYWEEKEND = case_when(
   DaysOfWeek %in%  6:7  ~ "WEEKEND"
 ))
 
+##Create DATE and MONTHDATE field for joining/plotting
+data$DATE <- as.Date(substr(data$DATETIME,1,10))
+data$YEARMONTH<- as.Date(paste(substr(data$DATE,1,7), "-01", sep=""))
+
+##Join solar and rainfall
+rainfall_nsw <- rainfall_nsw %>% 
+  unite(DATE, Year, Month, Day, sep="-") %>%
+  select(DATE, `Rainfall amount (millimetres)`) %>%
+  rename('DAILY_RAINFALL_AMT_MM'=`Rainfall amount (millimetres)`)
+
+solar_nsw <- solar_nsw %>% 
+  unite(DATE, Year, Month, Day, sep="-") %>%
+  select(DATE,`Daily global solar exposure (MJ/m*m)`) %>%
+  rename('DAILY_GLBL_SOLAR_EXPOSR'=`Daily global solar exposure (MJ/m*m)`)
+
+solar_nsw$DATE <- as.Date(solar_nsw$DATE)
+rainfall_nsw$DATE <- as.Date(rainfall_nsw$DATE)
+
+data <- data %>% 
+  left_join(rainfall_nsw,by=c('DATE')) %>%
+  left_join(solar_nsw,by=c('DATE'))
+
+
+##Deduplicate
+data <- data %>% distinct()
```

Comment:
why is there duplicates in the first place? best practice is to monitor each join in the first place to ensure no duplicates

Reply:
updated 👍